core/core-impl/src/main/java/ai/timefold/solver/core/impl/heuristic/selector/move/composite/UnionMoveSelectorFactory.java

```diff
@@ -22,8 +23,24 @@ public UnionMoveSelectorFactory(UnionMoveSelectorConfig moveSelectorConfig) {
     @Override
     protected MoveSelector<Solution_> buildBaseMoveSelector(HeuristicConfigPolicy<Solution_> configPolicy,
             SelectionCacheType minimumCacheType, boolean randomSelection) {
-        List<MoveSelector<Solution_>> moveSelectorList = buildInnerMoveSelectors(config.getMoveSelectorList(),
-                configPolicy, minimumCacheType, randomSelection);
+        List<MoveSelectorConfig> moveSelectorConfigList = new LinkedList<>(config.getMoveSelectorList());
+        if (configPolicy.getNearbyDistanceMeterClass() != null) {
+            for (MoveSelectorConfig selectorConfig : config.getMoveSelectorList().stream()
+                    .filter(MoveSelectorConfig::acceptNearbySelectionAutoConfiguration).toList()) {
+                if (selectorConfig.hasNearbySelectionConfig()) {
+                    throw new IllegalArgumentException(
+                            """
+                                    The selector configuration (%s) already includes the Nearby Selection setting, making it incompatible with the top-level property nearbyDistanceMeterClass (%s).
+                                    Remove the Nearby setting from the selector configuration or remove the top-level nearbyDistanceMeterClass."""
+                                    .formatted(selectorConfig, configPolicy.getNearbyDistanceMeterClass()));
+                }
+                // Add a new configuration with Nearby Selection enabled
+                moveSelectorConfigList.add(selectorConfig.enableNearbySelection(configPolicy.getNearbyDistanceMeterClass(),
+                        configPolicy.getRandom()));
+            }
+        }
+        List<MoveSelector<Solution_>> moveSelectorList =
```

Comment:
Cautious use of `var` wouldn't hurt here, I think.
The name already says `list`.